cpp/include/cudf/io/functions.hpp

```diff
@@ -238,6 +239,97 @@ table_with_metadata read_csv(
     read_csv_args const& args,
     rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+
+/**
+ * @brief Settings to use for `write_csv()`
+ */
+struct write_csv_args: private detail::csv::writer_options {
+  //inject private base accessors:
```

Comment:
Done.

Reply:
Man, I was going to fight for this ;-) Bring the big guns on this (Liskov substitution principle, creating artificial coupling between 2 classes that certainly are not in IS-A relationship, etc.). But, if this simplifies the cython / python side then...I submit.

It would be helpful if we collected such cython/python-side requirements for the C++ API in a gdoc or presentation.

Reply:
Yeah, I really liked the old design. Too bad.